openjdk.test.modularity/src/com.stf/net/adoptopenjdk/test/modularity/ExplicitModulesTest.java

```diff
@@ -0,0 +1,271 @@
+						.addArg("-d .")
+						.addArg( junitCopy.childDirectory("hamcrest.core/module-info.java").getSpec() + "  "));
+		
+		// Update the module info in hamcrest jar with module info file
+		test.doRunForegroundProcess("Updating the module info in hamcrest jar with generated module-info.class", "JAR", ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+					test.createJDKToolProcessDefinition()
+						.setJDKToolOrUtility("jar")	
+						.addArg("-uf " + junitCopy.childFile(explicitModule_HamcrestJarName).getSpec()  + " module-info.class" ));
+						
+		// Change into the junit directory inside temp and extract the junit jar as compiling module-info.java looks for the existence of packages.
+		test.doCd("Changing directory to thirdpartyjar/junit", junitCopy.childDirectory("junit") );
+		test.doRunForegroundProcess("Extracting junit jar", "JAR", ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+					test.createJDKToolProcessDefinition()
+						.setJDKToolOrUtility("jar")	
+						.addArg("-xf " + junitCopy.childFile(explicitModule_JunitJarName).getSpec() ));
+		
+		// Compile the module-info.java generated for junit-4.12.jar and update the module-info.class in the jar file.
+		test.doRunForegroundProcess("Compiling the module info", "JAVC", ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+					test.createJDKToolProcessDefinition()
+						.setJDKToolOrUtility("javac")	
+						.addArg("-d .")
+						.addArg("--module-path " + junitCopy.childDirectory("hamcrest.core").getSpec() )
+						.addArg( junitCopy.childDirectory("junit/module-info.java").getSpec() ));
+		
+		// Update the junit jar with generated module-info.java 
+		test.doRunForegroundProcess("Updating the junit jar with generated module-info.class", "JAR", ECHO_ON, ExpectedOutcome.cleanRun().within("1m"), 
+					test.createJDKToolProcessDefinition()
+						.setJDKToolOrUtility("jar")	
+						.addArg("-uf " + junitCopy.childFile(explicitModule_JunitJarName).getSpec() + " module-info.class" ));
+
+		// Usecase: 1.1
+		// Convert a non-modularized third party jar into an explicit module and verify if the application that uses it works fine.
+		// Here, the test App is an explicit module. Junit jar, when converted to explicit module, provides implicit readability to 
+		// hamcrest jar. The test App uses a type from hamcrest. The test App should be able to read hamcrest via implicit readability.
+		
+		// Currently, the behavior is that we are getting an IllegalAccessError on OpenJDK. 
+		// But, the test works fine on OpenJ9.
+		
+		//TODO: Failure of T1 on Oracle may be related to T4 failure (PR 134665) - In which case, it's a negative test
+		// and requires updates made to the use case. 
+		
+		String moduleName = "com.test";
+		String testName = "TestImpliedReadability";
+		String testClass = "com.ibm.runtimes.test." + testName;
+		String comment = "Run " + testName;
+
+		// Running com.test.TestImpliedReadaibility with junit jar and hamcrest jar as explicit modules.
```

Comment:
Cut'n'paste typo in comment TestImpliedReadability (rather than TestImpliedReadaibility), repeated several times in code below also.